crates/kitsune_p2p/types/src/timeout.rs

```diff
@@ -71,4 +123,19 @@ mod tests {
         assert!(t.time_remaining().as_micros() == 0);
         assert!(t.is_expired());
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn kitsune_backoff() {
+        let t = KitsuneTimeout::from_millis(100);
+        let mut times = Vec::new();
+        let start = std::time::Instant::now();
+        let bo = t.backoff(2, 15);
+        while !t.is_expired() {
+            times.push(start.elapsed().as_millis() as u64);
+            bo.wait().await;
+        }
+        println!("{:?}", times);
+        assert!(times.len() > 4);
```

Comment:
I'm surprised that `times` would be over 4. `2 + 4 + 8 + 16 + 32` is way over 15. Am I reading this wrong? Or are you *expecting* race conditions in the backoff for this test to work?

Reply:
so, yeah, the initial is `2` ms, the max is `15` ms, and we're letting the loop run for `100` ms, and seeing how many times are pushed on to the array... so, assuming that `.await` returns at exactly the backoff timeout without any additional delay, we'd get:

```rust
[
  0, // initial
  2, // after 2 ms backoff
  6, // after 4 ms backoff
  14, // after 8 ms backoff
  29, // after 15 ms backoff
  44, // after 15 ms backoff
  59, // after 15 ms backoff
  74, // after 15 ms backoff
  89, // after 15 ms backoff
].len() == 9 // entries
```